code/modules/projectiles/guns/flamer.dm

```diff
@@ -267,15 +267,10 @@
 				else
 					log_combat(user, H, "shot", src)
 
-			if(istype(H.wear_suit, /obj/item/clothing/suit/fire) || (istype(H.wear_suit, /obj/item/clothing/suit/storage/marine/M35) && istype(H.head, /obj/item/clothing/head/helmet/marine/pyro)))
+			if(H.IsFireResistant())
 				continue
 
 		var/armor_block = M.run_armor_check(null, "fire")
-		if(ishuman(M))
-			var/mob/living/carbon/human/H = M
-			if(istype(H.wear_suit, /obj/item/clothing/suit/fire) || (istype(H.wear_suit, /obj/item/clothing/suit/storage/marine/M35) && istype(H.head, /obj/item/clothing/head/helmet/marine/pyro)))
-				H.show_message(text("Your suit protects you from most of the flames."), 1)
-				armor_block = CLAMP(armor_block * 1.5, 0.75, 1) //Min 75% resist, max 100%
```

Comment:
There is a human check just above this, making this an unreachable code path

Reply:
Which part is unreachable here?

Reply:
Anything after the suit check, the suit check already happened above and continued on the loop.

Reply:
Aha, you are correct.